File: frontend/scratch/test-db.ts
// scratch/test-db.ts
import { db } from '../lib/db'
import { session as sessionTable } from '../lib/schema'

async function run() {
  try {
    console.log("Querying database...");
    const result = await db.select().from(sessionTable).limit(1);
    console.log("Database result:", result);
  } catch (error: any) {
    console.error("Database query failed!");
    console.error(error);
    if (error.cause) {
      console.error("Cause:", error.cause);
    }
  }
}

run();

File: backend/scratch/test_db.py
# backend/scratch/test_db.py
import os
import sys
from dotenv import load_dotenv
from sqlalchemy import create_engine, text

# Load environment
load_dotenv(os.path.join(os.path.dirname(__file__), '../.env'))

db_url = os.getenv("DATABASE_URL")
print("Raw Database URL:", db_url)

if db_url and db_url.startswith("postgresql://"):
    db_url = db_url.replace("postgresql://", "postgresql+psycopg2://", 1)

print("SQLAlchemy Database URL:", db_url)

try:
    print("Creating engine...")
    engine = create_engine(db_url, pool_pre_ping=True)
    
    print("Connecting to database...")
    with engine.connect() as conn:
        print("Executing test query...")
        result = conn.execute(text("SELECT 1")).scalar()
        print("Result of 'SELECT 1':", result)
        
except Exception as e:
    print("Connection failed!")
    print(e)
    # If psycopg2 operational error, get the cause/trace
    import traceback
    traceback.print_exc()
